### PR TITLE
Add GCC sample.

### LINKED ISSUE #58: Add sample gcc project to the repo

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #58.

### CODE REVIEW

**Review by arturdryomov — APPROVED** [review]

File: samples/gcc/.mainframer/remoteignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Usually you don't need to sync sources back from remote to local machine.
+*.c
```
Comment (gitanshu) [inline samples/gcc/.mainframer/remoteignore:2]: also ignore *.h i guess

Comment (artem-zinnatullin) [reply samples/gcc/.mainframer/remoteignore:2]: In this sample we don't have header files, but PR welcome :)

Comment (gitanshu) [reply samples/gcc/.mainframer/remoteignore:2]: Will do 😬

Comment (artem-zinnatullin) [reply samples/gcc/.mainframer/remoteignore:2]: Please take a look at https://github.com/github/gitignore/blob/master/C.gitignore, maybe it makes sense to include something from it.

**Pull request merged**